File: code/visualize/visualize_lig_prop.m
function [] = visualize_lig_prop(result, result_foldername)
%% Define color of bar
color_rec = {'#345995', '#eac435'};

%% Calculate proportion of free and bound ligand
VA = [result.VA_free(end), result.VA_bound(end)]/(result.VA_free(end) + result.VA_bound(end)) * 100;
VB = [result.VB_free(end), result.VB_bound(end)]/(result.VB_free(end) + result.VB_bound(end)) * 100;
Pl = [result.Pl_free(end), result.Pl_bound(end)]/(result.Pl_free(end) + result.Pl_bound(end)) * 100;
PDAA = [result.PDAA_free(end), result.PDAA_bound(end)]/(result.PDAA_free(end) + result.PDAA_bound(end)) * 100;
PDAB = [result.PDAB_free(end), result.PDAB_bound(end)]/(result.PDAB_free(end) + result.PDAB_bound(end)) * 100;
PDBB = [result.PDBB_free(end), result.PDBB_bound(end)]/(result.PDBB_free(end) + result.PDBB_bound(end)) * 100;

%% Generate data for bar plot
data = [VA; VB; Pl; PDAA; PDAB; PDBB];

%% Plot figure
figure('Position', [10 10 1200 500]);
b = bar(1:6, data, 'stacked', 'EdgeColor', 'none', 'BarWidth', 0.5);
b(1).FaceColor = color_rec{4};
b(2).FaceColor = color_rec{5};
for j = 1:size(data, 2)
    for i = 1:size(data, 1)
            if j ~= 2
                text(i-0.1, sum(data(i, 1:j))+3, sprintf('%.0f', data(i, j)), 'FontSize', 20, 'Color', 'k');
            else
                text(i-0.1, sum(data(i, 1:j))-5, sprintf('%.0f', data(i, j)), 'FontSize', 20, 'Color', 'k');
            end
    end
end
xlim([0 7]);
ylim([0 100]);
xticklabels({'VEGF-A', 'VEGF-B', 'PlGF', 'PDGF-AA', 'PDGF-AB', 'PDGF-BB'})
ylabel('Percentage (%)')
legend('Free', 'Bound', 'location', 'northeastoutside')
set(gca, 'fontsize', 25);
saveas(gca, sprintf('%s/Ligand_free_VS_bound', result_foldername), 'png');
saveas(gca, sprintf('%s/Ligand_free_VS_bound', result_foldername), 'epsc');

File: code_R/model/visualize_lig_prop.R
visualize_lig_prop <- function(results, result_foldername) {
  # Define colors -----------------------------------------------------------
  colors = c('#eac435', '#345995')
  
  # Calculate the percentage of free vs. bound ligand -----------------------
  lig = c('VEGF-A', 'VEGF-B', 'PlGF', 'PDGF-AA', 'PDGF-AB', 'PDGF-BB')
  lig_result = c('VA', 'VB', 'Pl', 'PDAA', 'PDAB', 'PDBB')
  
  df <- data.frame(matrix(ncol=3, nrow=0))
  colnames(df) <- c('Ligand', 'Status', 'Value')
  for (i in 1:length(lig)) {
    lig_name = lig[i]
    lig_result_name = lig_result[i]
    # Free ligand
    FreeColName = paste0(lig_result_name, '_free')
    # Bound ligand
    BoundColName = paste0(lig_result_name, '_bound')
    # Calculate the percentage of free and bound ligand
    value <- unname(
      tail(results[c(FreeColName, BoundColName)], 1)
      /sum(tail(results[c(FreeColName, BoundColName)], 1))
    )
    # Record
    new_df <- data.frame(Ligand=rep(lig_name, 2), 
                         Status=c('Free', 'Bound'), 
                         Value=as.vector(t(value)), 
                         row.names = NULL)
    # Append df
    df <- rbind(df, new_df)
  }
  # Fix the order of row
  df$Ligand <- factor(df$Ligand, levels=lig)

  # Plot the bar graph ------------------------------------------------------
  p = df %>% 
    mutate(percent_labels = scales::percent(Value, accuracy=1)) %>% 
    ggplot(aes(x = Ligand, y = Value, fill = Status)) +
    geom_col() +
    geom_text(
      aes(label = percent_labels), 
      position = position_stack(vjust = 0.5)
    ) + 
    scale_y_continuous(labels = scales::percent_format()) +
    scale_fill_manual(values=colors) +
    labs(x = "Ligand", y = "Percentage (%)") +
    theme(text = element_text(size= 10)) + theme_light()
  show(p)
  ggsave(sprintf("%s/Ligand_free_vs_bound.pdf", result_foldername), width = 6, height = 3, units = "in")
}